pkg/dryrun/dryrun.go

```diff
@@ -0,0 +1,810 @@
+			return nil, fmt.Errorf("require 1 object in the config file, found objects: %+v", objs)
+		}
+		cluster, ok := objs[0].(*corev1.StorageCluster)
+		if !ok {
+			return nil, fmt.Errorf("object is not storagecluster, %+v", objs[0])
+		}
+		return cluster, nil
+	}
+
+	if d.realClient == nil {
+		return nil, fmt.Errorf("could not load storage cluster, either pass an input file or KUBECONFIG")
+	}
+	nsList := &v1.NamespaceList{}
+	err := d.realClient.List(context.TODO(), nsList)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range nsList.Items {
+		clusterList := &corev1.StorageClusterList{}
+		err = d.realClient.List(context.TODO(), clusterList, &client.ListOptions{
+			Namespace: namespace.Name,
+		})
+		if err != nil {
+			logrus.WithError(err).Error("failed to list StorageCluster")
+		}
+		if len(clusterList.Items) > 0 {
+			cluster := &clusterList.Items[0]
+			d.cleanupObject(cluster)
+			return cluster, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not find storagecluster object from k8s")
+}
+
+func (d *DryRun) getRealK8sClient(kubeconfig string) (client.Client, error) {
+	var err error
+	var config *rest.Config
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+	d.outputFolder = defaultOutputFolder
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+		if err == nil {
+			// When running inside of container, creating folder on root will get permission denied.
+			d.outputFolder = "/tmp/" + d.outputFolder
+			logrus.Infof("output folder is %s", d.outputFolder)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	d.realControllerMgr, err = manager.New(config, manager.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := apis.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add resources to the scheme: %v", err)
+	}
+	if err := monitoringv1.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add prometheus resources to the scheme: %v", err)
+	}
+
+	if err := cluster_v1alpha1.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add cluster API resources to the scheme: %v", err)
+	}
+
+	if err := ocp_configv1.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add cluster API resources to the scheme: %v", err)
+	}
+
+	if err := corev1.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add corev1 resources to the scheme: %v", err)
+	}
+
+	if err := v1.AddToScheme(d.realControllerMgr.GetScheme()); err != nil {
+		logrus.Fatalf("Failed to add v1 resources to the scheme: %v", err)
+	}
+
+	start := func() {
+		if err := d.realControllerMgr.Start(ctrl.SetupSignalHandler()); err != nil {
+			logrus.Fatalf("Manager exited non-zero error: %v", err)
+		}
+	}
+	go start()
+	// Wait for the cache to be started.
+	time.Sleep(time.Second)
```

Comment:
Is 1 sec enough?

Reply:
yes, this should be good. it just needs to start without the cache to be fully populated.